src/index.js

```diff
@@ -279,42 +270,45 @@ function walk(exports, includeDefault) {
 function getMain({ options, entry, format }) {
 	const { pkg } = options;
 	const pkgMain = options['pkg-main'];
+	const pkgTypeModule = pkg.type === 'module';
+	let multipleEntries = options.multipleEntries;
 
 	if (!pkgMain) {
 		return options.output;
 	}
 
-	let mainNoExtension = options.output;
-	if (options.multipleEntries) {
+	let defaultOutput = options.output;
+	if (multipleEntries) {
 		let name = entry.match(new RegExp(/([\\/])index/.source + EXTENSION.source))
-			? mainNoExtension
+			? defaultOutput
 			: entry;
-		mainNoExtension = resolve(dirname(mainNoExtension), basename(name));
+		defaultOutput = resolve(dirname(defaultOutput), basename(name));
 	}
-	mainNoExtension = mainNoExtension.replace(EXTENSION, '');
+	const defaultOutputNoExtension = defaultOutput.replace(EXTENSION, '');
 
 	const mainsByFormat = {};
 
-	mainsByFormat.es = replaceName(
-		pkg.module && !pkg.module.match(/src\//)
-			? pkg.module
-			: pkg['jsnext:main'] || 'x.esm.js',
-		mainNoExtension,
+	mainsByFormat.es = resolve(
+		(!multipleEntries &&
+			(pkg.module && !pkg.module.match(/src\//)
```

Comment:
Random q: What's this match against `src/` about? Goes back to repo init afaict so couldn't find any context there.